pkg/pillar/cmd/volumemgr/initialvolumestatus.go

```diff
@@ -28,61 +28,54 @@ import (
 var nilUUID = uuid.UUID{}
 
 // appRwVolumeName - Returns name of the image ( including parent dir )
-// Note that we still use the sha in the filename to not impact running images. Otherwise
-// we could switch this to imageID
-// XXX other types of volumes might want a different name.
-func appRwVolumeName(sha256, uuidStr string, purgeCounter uint32, format zconfig.Format,
-	origin types.OriginType, isContainer bool) string {
+func appRwVolumeName(uuidStr string, generationCounter uint32,
+	format zconfig.Format, isContainer bool) string {
 
 	purgeString := ""
-	if purgeCounter != 0 {
-		purgeString = fmt.Sprintf("#%d", purgeCounter)
+	if generationCounter != 0 {
+		purgeString = fmt.Sprintf("#%d", generationCounter)
 	}
 	if isContainer {
-		return fmt.Sprintf("%s-%s%s", sha256, uuidStr, purgeString)
-	}
-	if origin != types.OriginTypeDownload {
-		log.Fatalf("XXX unsupported origin %v", origin)
+		return fmt.Sprintf("%s%s", uuidStr, purgeString)
 	}
 	formatStr := strings.ToLower(format.String())
-	return fmt.Sprintf("%s/%s-%s%s.%s", rwImgDirname, sha256,
+	return fmt.Sprintf("%s/%s%s.%s", rwImgDirname,
 		uuidStr, purgeString, formatStr)
 }
 
-// parseAppRwVolumeName - Returns rwImgDirname, sha256, uuidStr, purgeCounter
-func parseAppRwVolumeName(image string, isContainer bool) (string, string, string, uint32) {
+// parseAppRwVolumeName - Returns rwImgDirname, volume uuid, generationCounter
+func parseAppRwVolumeName(image string, isContainer bool) (string, string, uint32) {
 	// VolumeSha is provided by the controller - it can be uppercase
 	// or lowercase.
 	var re1 *regexp.Regexp
 	var re2 *regexp.Regexp
 	if isContainer {
-		re1 = regexp.MustCompile(`(.+)/([0-9A-Fa-f]+)-([0-9a-fA-F\-]+)#([0-9]+)`)
+		re1 = regexp.MustCompile(`(.+)/([0-9a-fA-F\-]+)#([0-9]+)`)
 	} else {
-		re1 = regexp.MustCompile(`(.+)/([0-9A-Fa-f]+)-([0-9a-fA-F\-]+)#([0-9]+)\.(.+)`)
+		re1 = regexp.MustCompile(`(.+)/([0-9a-fA-F\-]+)#([0-9]+)\.(.+)`)
```

Comment:
Note that this is impacted by the suggestion above.

Reply:
Noted.